process/smartContract/scQueryService_test.go

```diff
@@ -457,7 +464,97 @@ func TestExecuteQuery_ShouldReceiveQueryCorrectly(t *testing.T) {
 		assert.True(t, wasRecreateTrieCalled)
 		assert.Nil(t, err)
 	})
-	t.Run("block nonce should work", func(t *testing.T) {
+	t.Run("block hash should work - current epoch", func(t *testing.T) {
+		t.Parallel()
+
+		runWasCalled := false
+
+		mockVM := &mock.VMExecutionHandlerStub{
+			RunSmartContractCallCalled: func(input *vmcommon.ContractCallInput) (output *vmcommon.VMOutput, e error) {
+				runWasCalled = true
+				assert.Equal(t, int64(42), big.NewInt(0).SetBytes(input.Arguments[0]).Int64())
+				assert.Equal(t, int64(43), big.NewInt(0).SetBytes(input.Arguments[1]).Int64())
+				assert.Equal(t, scAddress, input.CallerAddr)
+				assert.Equal(t, funcName, input.Function)
+
+				return &vmcommon.VMOutput{
+					ReturnCode: vmcommon.Ok,
+				}, nil
+			},
+		}
+		argsNewSCQuery := createMockArgumentsForSCQuery()
+		argsNewSCQuery.VmContainer = &mock.VMContainerMock{
+			GetCalled: func(key []byte) (handler vmcommon.VMExecutionHandler, e error) {
+				return mockVM, nil
+			},
+		}
+		argsNewSCQuery.EconomicsFee = &economicsmocks.EconomicsHandlerStub{
+			MaxGasLimitPerBlockCalled: func(_ uint32) uint64 {
+				return uint64(math.MaxUint64)
+			},
+		}
+		providedHash := []byte("provided hash")
+		providedRootHash := []byte("provided root hash")
+		argsNewSCQuery.Marshaller = &marshallerMock.MarshalizerMock{}
+		counter := 0
```

Comment:
Can be renamed to `getFromEpochCallsCounter` (if makes sense).

Oh, now I see, can be left as it is - seen in other tests, as well.

Reply:
removed, anyway